src/client/clientlauncher.cpp

```diff
@@ -59,7 +59,8 @@ ClientLauncher::~ClientLauncher()
 
 	delete g_fontengine;
 
-	device->drop();
+	if (device)
```

Comment:
* Does this adjustment indicate that your first version for a pull request around pointer checks should be rejected?
* Will another pull request be attempted which contains only acceptable changes?